### PR TITLE
Update: add support for semicolon in comma-first setup in indent rule

### LINKED ISSUE #3423: semicolon in comma-first setup regression from 0.24.1
I got following issue (in 1.1.0, worked perfectly in 0.24.1):

eslint config:

```
    "indent": [2, 2]
```

code:

```
11 var path     = require('path')
12  , crypto    = require('crypto')
13  ;
```

output:

```
   13:4  error  Expected indentation of 0 characters but found 2  indent
```

Thank you.

PS. Discussed in greater details here: https://github.com/eslint/eslint/issues/3391

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3423

### CODE REVIEW

**Comment by gyandeeps** [comment]
Can you add unit test for these use cases:"

```js
// Case 1:
var a = 1
   ,b = 2
   ;
   
// Case 2:
var a = 1
   ,b = 2
 ;
```

File: tests/lib/rules/indent.js [hunk 1]
```diff
@@ -682,6 +682,12 @@ ruleTester.run("indent", rule, {
                 "  }\n" +
                 "};",
             options: [2]
+        }, {
+            code:
+                "var path     = require('path')\n" +
+                " , crypto    = require('crypto')\n" +
+                " ;\n",
```
Comment (gyandeeps) [inline tests/lib/rules/indent.js]: Why is this unit test passing?
`crypto` has 3 spaces before it? I expect it to have 4.

**Comment by gyandeeps** [comment]
Right now this rule is not considering comma before variable declarations. To do that we have to do something similar to what is going on for check for array or object indentation. https://github.com/eslint/eslint/blob/master/lib/rules/indent.js#L378

We have to know if the comma is first then we have to pass in `true` as the last parameter so that it knows to ignore commas when evaluating spaces.

File: tests/lib/rules/indent.js (hunk 1 above)
Comment (BYK) [inline tests/lib/rules/indent.js]: No, it has `2` spaces before it (excluding comma) and the config also says `2` spaces so I think this is normal. Also, even if that part was unexpected, this patch only fixes the behavior for the semicolon, not anything else.

**Comment by BYK** [comment]
>Right now this rule is not considering comma before variable declarations.

I'm not sure why you thought I was addressing this. I only enforce the semicolon to be at the same indent with the comma if the previous line starts with a comma, that's about it.

**Comment by BYK** [comment]
@gyandeeps - Added the test cases you requested.

File: tests/lib/rules/indent.js (hunk 1 above)
Comment (gyandeeps) [inline tests/lib/rules/indent.js]: lol. For got about the 2 spaces. I had 4 in my head.

File: tests/lib/rules/indent.js [hunk 2]
```diff
@@ -682,6 +682,17 @@ ruleTester.run("indent", rule, {
                 "  }\n" +
                 "};",
             options: [2]
+        }, {
```
Comment (gyandeeps) [inline tests/lib/rules/indent.js]: Just to be consistent can you please put braces on next line.

**Pull request merged**